string type
string md5sum
string req_type
string res_type
